chain/rosetta-rpc/src/adapters/nep141.rs

```diff
@@ -0,0 +1,209 @@
+use std::{collections::HashMap, str::FromStr};
+
+use near_o11y::WithSpanContextExt;
+use near_primitives::{types::BlockId, views::ExecutionOutcomeWithIdView};
+
+use crate::models::{AccountIdentifier, Currency, FungibleTokenEvent};
+
+pub(crate) async fn collect_nep141_events(
+    receipt_execution_outcomes: &Vec<ExecutionOutcomeWithIdView>,
+    block_header: &near_primitives::views::BlockHeaderView,
+    currencies: &Option<Vec<Currency>>,
+) -> crate::errors::Result<Vec<FungibleTokenEvent>> {
+    let mut res = Vec::new();
+    for outcome in receipt_execution_outcomes {
+        let events = extract_events(outcome);
+        for event in events {
+            res.extend(
+                compose_rosetta_nep141_events(&event, outcome, block_header, currencies).await?,
+            );
+        }
+    }
+    Ok(res)
+}
+
+async fn compose_rosetta_nep141_events(
+    events: &crate::models::Nep141Event,
+    outcome: &ExecutionOutcomeWithIdView,
+    block_header: &near_primitives::views::BlockHeaderView,
+    currencies: &Option<Vec<Currency>>,
+) -> crate::errors::Result<Vec<FungibleTokenEvent>> {
+    let mut ft_events = Vec::new();
+    match &events.event_kind {
+        crate::models::Nep141EventKind::FtTransfer(transfer_events) => {
+            if let Some(currencies) = currencies {
+                let currency_map: std::collections::HashMap<String, Currency> =
+                    currencies.clone().into_iter().collect::<HashMap<String, Currency>>();
```

Comment:
(optional) nit: you should be able to avoid the `Vec::clone` and instead only clone the strings.
```suggestion
                    currencies.iter().cloned().collect::<HashMap<String, Currency>>();
```